DataFormats/HGCDigi/interface/PreMixHGCSimAccumulator.h

```diff
@@ -0,0 +1,162 @@
+#ifndef DataFormats_HGCDigi_PreMixHGCSimAccumulator_h
+#define DataFormats_HGCDigi_PreMixHGCSimAccumulator_h
+
+#include "DataFormats/DetId/interface/DetId.h"
+#include <iostream>
+#include <vector>
+#include <cassert>
+
+class PreMixHGCSimAccumulator {
+public:
+  class DetIdSize {
+  public:
+    DetIdSize() {}
+    DetIdSize(unsigned int detId) : detId_(detId) {}
+    void increaseSize() { ++size_; }
+    unsigned int detId() const { return detId_; }
+    unsigned int size() const { return size_; }
+
+  private:
+    unsigned int detId_ = 0;
+    unsigned char size_ = 0;
+  };
+
+  class SimHitCollection {
+  public:
+    constexpr static unsigned energyOffset = 15;
+    constexpr static unsigned energyMask = 0x1;
+    constexpr static unsigned sampleOffset = 11;
+    constexpr static unsigned sampleMask = 0xf;
+    constexpr static unsigned dataOffset = 0;
+    constexpr static unsigned dataMask = 0x7ff;
+
+    SimHitCollection() {}
+    SimHitCollection(unsigned int nhits) : nhits_(nhits) {}
+    SimHitCollection(const unsigned short si,
+                     const std::vector<unsigned short>& accCharge,
+                     const std::vector<unsigned short>& time)
+        : nhits_(accCharge.size()) {
+      //std::cout<<"### printing from PreMix "<<nhits_<<std::endl;
+      chargeArray_.reserve(nhits_);
+      timeArray_.reserve(nhits_);
+      for (size_t i = 0; i < nhits_; ++i) {
+        unsigned short ei = 0;
+        unsigned short d = accCharge[i];
+        unsigned short data = ((ei << energyOffset) | (si << sampleOffset) | d);
+        chargeArray_.emplace_back(data);
+      }
+      for (size_t i = 0; i < nhits_; ++i) {
+        unsigned short ei = 1;
+        unsigned short d = time[i];
+        unsigned short data = ((ei << energyOffset) | (si << sampleOffset) | d);
+        timeArray_.emplace_back(data);
+      }
+    }
+    SimHitCollection(const SimHitCollection& simhitcollection)
+        : nhits_(simhitcollection.nhits_),
+          chargeArray_(simhitcollection.chargeArray_),
+          timeArray_(simhitcollection.timeArray_) {}
+    unsigned int nhits() const { return nhits_; }
+    unsigned int sampleIndex() const {
+      //std::cout<<chargeArray_[0]<<"\t"<<sampleOffset<<std::endl;
+      return (chargeArray_[0] >> sampleOffset) & sampleMask;
+    }
+    std::vector<unsigned short> chargeArray() const { return chargeArray_; }
+    std::vector<unsigned short> timeArray() const { return timeArray_; }
```

Comment:
```suggestion
    const std::vector<unsigned short>& chargeArray() const { return chargeArray_; }
    const std::vector<unsigned short>& timeArray() const { return timeArray_; }
```
to avoid unnecessary copying